xbmc/cores/VideoPlayer/DVDMessageQueue.cpp

```diff
@@ -150,9 +169,11 @@ MsgQueueReturnCode CDVDMessageQueue::Get(CDVDMsg** pMsg, unsigned int iTimeoutIn
 
   while (!m_bAbortRequest)
   {
-    if (!m_list.empty() && m_list.back().priority >= priority)
+    std::list<DVDMessageListItem> &msgs = (priority > 0 || !m_prioMessages.empty()) ? m_prioMessages : m_messages;
```

Comment:
This looks a bit strange to me. Using ``&&`` rather than ``||`` here would make more sense to me.
I may well be missing something, so feel free to ignore if you are happy with it.